File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadOutlookBar/OutlookBarContentTemplateSelector.cs
using System.Windows;
using System.Windows.Controls;

namespace OpenSilver.Samples.TelerikUI.RadOutlookBar
{
    public class OutlookBarContentTemplateSelector : DataTemplateSelector
    {
        public DataTemplate MailTemplate { get; set; }
        public DataTemplate CalendarTemplate { get; set; }
        public DataTemplate ContactsTemplate { get; set; }

        public override DataTemplate SelectTemplate(object item, DependencyObject container)
        {
            if (item is MailMenuItem)
            {
                return MailTemplate;
            }
            else if (item is CalendarMenuItem)
            {
                return CalendarTemplate;
            }
            else if (item is ContactsMenuItem)
            {
                return ContactsTemplate;
            }

            return base.SelectTemplate(item, container);
        }
    }
}

File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadOutlookBar/OutlookBarContentTemplateSelector.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace OpenSilver.Samples.TelerikUI.RadOutlookBar
    Public Class OutlookBarContentTemplateSelector
        Inherits DataTemplateSelector
        Public Property MailTemplate As DataTemplate
        Public Property CalendarTemplate As DataTemplate
        Public Property ContactsTemplate As DataTemplate

        Public Overrides Function SelectTemplate(item As Object, container As DependencyObject) As DataTemplate
            If TypeOf item Is MailMenuItem Then
                Return MailTemplate
            ElseIf TypeOf item Is CalendarMenuItem Then
                Return CalendarTemplate
            ElseIf TypeOf item Is ContactsMenuItem Then
                Return ContactsTemplate
            End If

            Return MyBase.SelectTemplate(item, container)
        End Function
    End Class
End Namespace
